buildSrc/src/main/kotlin/PluginInfoCollectorPlugin.kt

```diff
@@ -0,0 +1,153 @@
+     */
+    private fun configureCollectDependenciesTask(project: Project, task: CollectDependencyPluginsTask) {
+        val jarFiles = mutableListOf<File>()
+
+        // The compileOnly configuration cannot be resolved; so a detached configuration is created.
+        project.configurations.findByName(DEPENDENCY_CONFIGURATION_NAME)?.also { configuration ->
+            val detachedConfig = project.configurations.detachedConfiguration()
+            configuration.dependencies.forEach { dep ->
+                detachedConfig.dependencies.add(dep)
+
+                (dep as? ProjectDependency)?.also { projectDependency ->
+                    project.gradle.projectsEvaluated {
+                        val jarTask = project.project(projectDependency.path)
+                            .tasks.named("jar", Jar::class.java)
+                        task.jarFiles.from(jarTask.get().archiveFile)
+                        task.dependsOn(jarTask)
+                    }
+                }
+            }
+
+            detachedConfig.resolvedConfiguration.resolvedArtifacts.forEach { artifact ->
+                if (artifact.file.extension == "jar") {
+                    jarFiles.add(artifact.file)
+                }
+            }
+        }
+
+        task.jarFiles.setFrom(jarFiles)
+    }
+}
+
+/**
+ * A task that iterates over the dependencies of the current project and extracts information about all ORT plugins
+ * that are referenced by them.
+ */
+abstract class CollectDependencyPluginsTask : DefaultTask() {
+    companion object {
+        /** Regular expression to match plugin information files in JARs. */
+        private val regexPluginFile = Regex("META-INF/plugin/.*\\.json")
+    }
+
+    @get:InputFiles
+    @get:PathSensitive(PathSensitivity.NONE)
+    abstract val jarFiles: ConfigurableFileCollection
+
+    /** The directory where the collected plugin information files will be stored. */
+    @get:OutputDirectory
+    abstract val outputDirectory: DirectoryProperty
+
+    @TaskAction
+    fun findPluginInfosInDependencies() {
+        val extractDir = outputDirectory.get().asFile
+        extractDir.deleteRecursively()
+        extractDir.mkdirs()
+
+        jarFiles.forEach { jarFile ->
+            extractPluginInfosFromJar(jarFile, extractDir)
+        }
+    }
+
+    /**
+     * Read the given [jarFile] and extract all information files about ORT plugins to the given [extractDir].
+     */
+    private fun extractPluginInfosFromJar(jarFile: File, extractDir: File) {
+        try {
+            ZipFile(jarFile).use { zip ->
+                zip.entries().asSequence()
+                    .filter { entry ->
+                        !entry.isDirectory && entry.name.matches(regexPluginFile)
+                    }.forEach { entry ->
+                        val entryFile = File(entry.name)
+                        val targetFile = extractDir.resolve(entryFile.name)
+                        targetFile.parentFile.mkdirs()
```

Comment:
Is it not sufficient that `extractDir.mkdirs()` is already called above?

Reply:
Yes, dropped. Initially, files were collected in different folder structures derived from their origins.